### PR TITLE
Do not fail fetching service params for optional service

### PR DESCRIPTION
**Pull request by radito3**
#### Description: 
Do not fail when service parameters fetching fails for optional resources

### CODE REVIEW

File: multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java [hunk 1]
```diff
@@ -71,6 +75,31 @@ void testExecute(String inputFilename, String expectedExceptionMessage) {
...
+            step.execute(execution);
+            assertStepIsRunning();
+            List<ServiceAction> serviceActionsToExecute = context.getVariable(Variables.SERVICE_ACTIONS_TO_EXCECUTE);
+            assertTrue(serviceActionsToExecute.contains(ServiceAction.UPDATE_CREDENTIALS),
+                       "Actions should contain " + ServiceAction.UPDATE_CREDENTIALS);
+            return;
+        }
+        SLException exception = assertThrows(SLException.class, () -> step.execute(execution));
+        assertTrue(exception.getCause() instanceof CloudOperationException);
+        assertSame(((CloudOperationException) exception.getCause()).getStatusCode(), HttpStatus.INTERNAL_SERVER_ERROR);
```
Comment (theghost5800) [inline multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java:100]: Why don't use `assertEquals`?

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java [hunk 2]
```diff
@@ -272,7 +272,7 @@ private boolean shouldUpdateCredentials(CloudServiceInstanceExtended service, Cl
             getStepLogger().debug("Existing service parameters: " + SecureSerialization.toJson(serviceParameters));
             return !Objects.equals(service.getCredentials(), serviceParameters);
         } catch (CloudOperationException e) {
-            if (e.getStatusCode() == HttpStatus.BAD_REQUEST) {
+            if (e.getStatusCode() == HttpStatus.BAD_REQUEST || service.isOptional()) {
                 getStepLogger().warnWithoutProgressMessage(Messages.CANNOT_RETRIEVE_SERVICE_INSTANCE_PARAMETERS, service.getName());
                 // TODO: Optimization (Hack) that should be deprecated at some point. So here is a todo for that.
                 return !MapUtils.isEmpty(service.getCredentials());
```
Comment (theghost5800) [inline multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java:278]: What will happen if the service is optional and has specified service parameters and at this line return code is `true`? 
Is this mean that service parameters update will be triggered? How will be handled service parameters update when service instance doesn't exist?

File: multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java (hunk 1 above)
Comment (radito3) [reply multiapps-controller-process/src/test/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStepTest.java:100]: The status code is an enum variable, not an object. The `==` operator is sufficient for checking equality. This method was a suggestion by the IDE.

File: multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java (hunk 2 above)
Comment (radito3) [reply multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/steps/DetermineServiceCreateUpdateServiceActionsStep.java:278]: This code won't be executed if a service instance doesn't exist.

**Review by theghost5800 — APPROVED** [review]

**Pull request merged**